lib/react.dart

```diff
@@ -590,6 +591,14 @@ abstract class Component2 implements Component {
     adapter.setStateWithUpdater(updater, callback);
   }
 
+  /// Method used to initialize the [state] of a component.
+  ///
+  /// To avoid unnecessary Map copies and setter calls, the [init] method should use [initializeState] (instead of
+  /// this.state) to directly set the state on the JavaScript side.
```

Comment:
I think this is a little too much implementation detail for consumers.

An example might also be nice.

 What about something like:
```dart
/// Whereas `this.state` can be set directly in JavaScript components during initialization, 
/// supporting that is difficult in the Dart implementation, so this method must be used instead.
///
/// ```js
/// // JavaScript
/// constructor() {
///   this.state = {count: 0};
/// }
/// ```
///
/// ```dart
/// // Dart
/// init() {
///   initializeState({'count': 0});
/// }
```